app/src/main/java/com/github/mik629/android/fundamentals/ui/movieslist/FragmentMoviesList.kt

```diff
@@ -1,72 +1,78 @@
-    private lateinit var binding: FragmentMoviesListBinding
+class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
+    private val binding by viewBinding(FragmentMoviesListBinding::bind)
 
-    private val viewModel by lazy {
-        MoviesListViewModel(
-            MoviesRepositoryImpl(
-                AppModule().retrofit.create(ServerApi::class.java),
-                ActorMapper()
-            )
-        )
-    }
+    private val viewModel: MoviesListViewModel by viewModels(
+        factoryProducer = {
+            MoviesListViewModelFactory()
+        }
+    )
 
     private val movieItemAdapter by lazy {
         MovieItemAdapter(
-            { movieTitle ->
-                requireFragmentManager()
+            { movie ->
+                parentFragmentManager
                     .beginTransaction()
-                    .addToBackStack(FragmentMoviesList::class.simpleName)
-                    .add(R.id.main_container, FragmentMovieDetails.newInstance(movieTitle))
+                    .addToBackStack(FragmentMoviesList::class.qualifiedName) // fixme add cicerone
+                    .add(R.id.main_container, FragmentMovieDetails.newInstance(movie.id))
                     .commit()
             },
-            GlideApp.with(this)
-                .asDrawable()
-                .thumbnail(0.1f)
-                .placeholder(R.drawable.ic_image_loading)
-                .fallback(R.drawable.ic_broken_image)
-                .transition(DrawableTransitionOptions.withCrossFade())
-                .fitCenter()
+            buildGlideRequest(this)
         )
     }
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        binding = FragmentMoviesListBinding.inflate(layoutInflater)
-
-        with(binding) {
-            movieList.adapter = movieItemAdapter
-            viewModel.movies.observe(this@FragmentMoviesList.viewLifecycleOwner) {
-                movieItemAdapter.submitList(it)
-            }
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        viewModel.movies.observe(this@FragmentMoviesList.viewLifecycleOwner) {
+            movieItemAdapter.submitList(it)
+        }
+        viewModel.error.observe(this@FragmentMoviesList.viewLifecycleOwner) {
+            Snackbar.make(binding.root, getString(R.string.error_no_data), LENGTH_LONG)
+                .show()
         }
+    }
 
-        return binding.root
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.movieList.adapter = movieItemAdapter
+        binding.movieList.layoutManager = GridLayoutManager(
+            requireContext(),
+            resources.getInteger(R.integer.span_count),
+            resources.getInteger(R.integer.scroll_orientation),
```

Comment:
u have different orientation for GridLayoutManager for different screen orientations?

Reply:
Yeah, is it bad?

Reply:
it depends
if u think that this UX is OK that keep it

Reply:
I'm not an expert, but as a user I like it better